src/base-client.ts

```diff
@@ -1,169 +1,192 @@
+import { ConsoleLogger, Logger } from './logger';
 import { TokenProvider } from './token-provider';
-import { Subscription, SubscribeOptions } from './subscription';
 import { ResumableSubscribeOptions, ResumableSubscription} from './resumable-subscription';
 import { RetryStrategy, RetryStrategyResult, Retry, DoNotRetry, ExponentialBackoffRetryStrategy } from './retry-strategy';
+import { StatelessSubscribeOptions, StatelessSubscription} from './stateless-subscription';
 
 export interface BaseClientOptions {
     host: string;
-
+    
     encrypted?: boolean;
     timeout?: number;
     XMLHttpRequest?: Function;
+    logger?: Logger;
 }
 
 export type Headers = {
-  [key: string]: string;
-}
-
-export interface Event {
-  eventId: string;
-  headers: Headers;
-  body: any;
+    [key: string]: string;
 }
 
 export interface RequestOptions {
-  method: string;
-  path: string;
-  tokenProvider?: TokenProvider;
-  jwt?: string;
-  headers?: Headers;
-  body?: any;
-  retryStrategy?: RetryStrategy;
+    method: string;
+    path: string;
+    tokenProvider?: TokenProvider;
+    jwt?: string;
```

Comment:
I've added one - although the `createXHR` method uses it so I'll leave it in for now.